src/main/java/uk/gov/hmcts/reform/unspec/service/flowstate/FlowPredicate.java

```diff
@@ -6,14 +6,25 @@
 import java.util.function.Predicate;
 
 import static uk.gov.hmcts.reform.unspec.enums.CaseState.CLOSED;
+import static uk.gov.hmcts.reform.unspec.enums.CaseState.PENDING_CREATED;
 import static uk.gov.hmcts.reform.unspec.enums.CaseState.STAYED;
 
 public class FlowPredicate {
 
-    public static final Predicate<CaseData> claimantIssueClaim = caseData ->
+    public static final Predicate<CaseData> pendingCreated = caseData ->
         caseData.getClaimIssuedDate() != null
             && caseData.getLegacyCaseReference() != null;
 
+    public static final Predicate<CaseData> paymentFailed = caseData ->
+        caseData.getPaymentErrorMessage() != null;
```

Comment:
StateFlow should transition into `PAYMENT_FAILED` only if payment fails because of a business exception. I am concerned that there is a very slight chance we could get this wrong here if for any reason F&P API would return an empty error message in the event of an HTTPS ERROR CODE 403. Although very unlikely I believe it would be safer to introduce a new case field describing the payment status explicitly. That field would be set explicitly in PaymentsCallbackHandler.java 

Maybe we should also consider introducing a complex type to group all things related to PBA payment:

- Payment
  - Status: [Success|Failed]
  - ErrorCode
  - ErrorMessage

Reply:
I'd say we should trust F&P team in what they provide on their API - if 403 means that there will be error code and message, there should be no need for additional protection on our side.

But if you take a look in PaymentsCallbackHandler you'll see that for 403 error we're mapping the response to PaymentDto - and if it fails, then it's treated as a technical exception. We can add additional `notEmpty` check there, but I don't see any value in storing payment status separaately - error code and message are removed once the payment is successful.